go-controller/pkg/ovn/master.go

```diff
@@ -827,6 +847,52 @@ func (oc *Controller) clearInitialNodeNetworkUnavailableCondition(origNode, newN
 	}
 }
 
+// this is the worker function that does the periodic sync of nodes from kube API
+// and sbdb and deletes chassis that are stale
+
+func (oc *Controller) syncNodesPeriodic() {
+	//node names is a slice of all node names
+	nodeNames := make([]string, 0)
+	nodes, err := oc.kube.GetNodes()
+	if err != nil {
+		klog.Errorf("Error getting existing nodes from kube API: %v", err)
+		return
+	}
+
+	for _, node := range nodes.Items {
+		nodeNames = append(nodeNames, node.Name)
```

Comment:
this is going to be pretty slow since the initial capacity of nodeNames is 0, there's going to be a lot of new arrays created. Could you set initial capacity to be the same length as node.Items and do initialization just before this for loop runs?

Reply:
Good point. I can initialize it to length of `nodes.Items`